libtransmission/utils.cc

```diff
@@ -76,6 +76,15 @@ Config::Units<StorageUnits> Config::Storage{ Config::Base::Kilo, "B"sv, "kB"sv,
 
 // ---
 
+std::filesystem::path tr_get_lock_name(std::string const& config_dir)
+{
+    struct stat sb = {};
+    (void)::stat(config_dir.c_str(), &sb);
+    std::string filename = fmt::format("transmission_lock_{}_{}", sb.st_dev, sb.st_ino);
+
+    return std::filesystem::temp_directory_path() / filename;
```

Comment:
My intention was to place locks to volatile storage and make them always unlocked on start of OS. But now I think you are right, because a path to a temporary directory can be overriden by environment and be different for different instances. So lock files inside configuration directories is the right way to implement this.